statics/js/components/graph-layout.js

```diff
@@ -1269,45 +1197,60 @@ TopologyGraphLayout.prototype = {
+  },
+
+  updateLinkLabelData: function() {
     var self = this;
-    var bandwidth = this.bandwidth, defaultInterfaceSpeed = 1048576;
 
-    var i, link, links = this.links;
-    for (i in links) {
-      link = links[i];
-
-      if (link.metadata.RelationType != "layer2" || !link.target.metadata.LastUpdateMetric) continue;
-      if (!link.source.visible || !link.target.visible) continue;
-
-      var speed = (this.bandwidth.bandwidthThreshold === 'relative') ?
-        link.target.metadata.Speed || defaultInterfaceSpeed : 1;
-
-      var kbps = this.bandwidthFromMetrics(link.target.metadata.LastUpdateMetric);
-      if (link.target.metadata.LastUpdateMetric && kbps > bandwidth.active * speed) {
-        this.linkLabels[link.id] = {
-          "id": "link-label-" + link.id,
-          "link": link,
-          "text": bandwidthToString(kbps),
-          "active": (kbps > bandwidth.active * speed) && (kbps < bandwidth.warning * speed),
-          "warning": (kbps >= bandwidth.warning * speed) && (kbps < bandwidth.alert * speed),
-          "alert": kbps >= bandwidth.alert * speed
+    for (var i in this.links) {
+      var bandwidth = this.bandwidth;
+      var link = this.links[i];
+
+      if (!link.source.visible || !link.target.visible)
+        continue;
+      if (link.metadata.RelationType != "layer2")
+	continue;
+      if (!link.target.metadata.LastUpdateMetric)
+        continue;
+
+      const defaultBandwidthBaseline = 1024 * 1024 * 1024; // 1 gbps
+      var bandwidthBaseline = (bandwidth.bandwidthThreshold === 'relative') ?
+        link.target.metadata.Speed || defaultBandwidthBaseline : 1;
+      var bandwidthAbsolute = this.bandwidthFromMetrics(link.target.metadata.LastUpdateMetric);
+      var bandwidthCheck = bandwidthAbsolute / bandwidthBaseline;
+
+      if (bandwidthCheck > bandwidth.active) {
+        this.linkLabelData[link.id] = {
+          id: "link-label-" + link.id,
+          link: link,
+          text: bandwidthToString(bandwidthAbsolute),
+          active: (bandwidthCheck > bandwidth.active) && (bandwidthCheck < bandwidth.warning),
+          warning: (bandwidthCheck >= bandwidth.warning) && (bandwidthCheck < bandwidth.alert),
+          alert: bandwidthCheck >= bandwidth.alert
         };
       } else {
-        delete this.linkLabels[link.id];
+	delete this.linkLabelData[link.id];
       }
     }
+  },
+
+  updateBandwidth: function() {
+    var self = this;
 
-    this.linkLabel = this.linkLabel.data(Object.values(this.linkLabels), function(d) { return d.id; });
+    this.updateLinkLabelData()
+    this.updateLinkLabel()
     var exit = this.linkLabel.exit();
 
+    // update links which don't have traffic
     exit.each(function(d) {
       self.g.select("#link-" + d.link.id)
       .classed ("link-label-active", false)
       .classed ("link-label-warning", false)
       .classed ("link-label-alert", false);
     })
-    .remove();
+    exit.remove();
```

Comment:
a ";" is missing the previous line or this "exit" has to be removed

Reply:
fixed